src/sage/combinat/designs/designs_pyx.pyx

```diff
@@ -15,6 +15,202 @@ from cysignals.memory cimport sig_malloc, sig_calloc, sig_realloc, sig_free
+        ....:      [2, 0, 1, 2, 2, 1, 2, 0, 2, 0, 0, 1, 1],
+        ....:      [2, 2, 0, 1, 2, 2, 1, 2, 0, 2, 0, 0, 1],
+        ....:      [2, 2, 2, 0, 1, 2, 2, 1, 2, 0, 2, 0, 0],
+        ....:      [0, 2, 2, 2, 0, 1, 2, 2, 1, 2, 0, 2, 0],
+        ....:      [0, 0, 2, 2, 2, 0, 1, 2, 2, 1, 2, 0, 2],
+        ....:      [1, 0, 0, 2, 2, 2, 0, 1, 2, 2, 1, 2, 0],
+        ....:      [0, 1, 0, 0, 2, 2, 2, 0, 1, 2, 2, 1, 2],
+        ....:      [1, 0, 1, 0, 0, 2, 2, 2, 0, 1, 2, 2, 1],
+        ....:      [2, 1, 0, 1, 0, 0, 2, 2, 2, 0, 1, 2, 2],
+        ....:      [1, 2, 1, 0, 1, 0, 0, 2, 2, 2, 0, 1, 2],
+        ....:      [1, 1, 2, 1, 0, 1, 0, 0, 2, 2, 2, 0, 1],
+        ....:      [2, 1, 1, 2, 1, 0, 1, 0, 0, 2, 2, 2, 0],
+        ....:      [0, 2, 1, 1, 2, 1, 0, 1, 0, 0, 2, 2, 2],
+        ....:      [1, 0, 2, 1, 1, 2, 1, 0, 1, 0, 0, 2, 2],
+        ....:      [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]
+        sage: is_covering_array(C,parameters=True)
+        (True, (53, 3, 13, 3))
+
+        sage: C = [[1, 0, 1, 1, 2, 0, 2, 2],
+        ....:      [2, 1, 0, 1, 1, 2, 0, 2],
+        ....:      [2, 2, 1, 0, 1, 1, 2, 0],
+        ....:      [0, 2, 2, 1, 0, 1, 1, 2],
+        ....:      [2, 0, 2, 2, 1, 0, 1, 1],
+        ....:      [1, 2, 0, 2, 2, 1, 0, 1],
+        ....:      [1, 1, 2, 0, 2, 2, 1, 0],
+        ....:      [0, 1, 1, 2, 0, 2, 2, 1]]
+        sage: is_covering_array(C,strength=2,parameters=True)
+        (False, (8, 0, 8, 3))
+
+    """
+    from itertools import product, combinations
+
+    if levels is None:
+        symbol_list = list({x for l in array for x in l})
+        levels = len(symbol_list)
+    else:
+        symbol_list = [num for num in range(levels)]
+
+    number_rows = len(array)
+    number_columns = len(array[0])
+
+    # Set wstrength to be the current t value to be checked
+    if strength is None:
+        wstrength = 1
+    else:
+        if strength > number_columns:
+            raise ValueError("Strength must be equal or less than number of columns")
+        wstrength = strength
+
+    for row in array:
+        if len(row) != number_columns:
+            raise ValueError("Not all rows are the same length, row {} is not the same length as row 0".format(array.index(row)))
+        else:
+            for entry in row:
+                if int(entry) != entry or entry < -1 or entry >= levels:
+                    raise ValueError("Array should contain integer symbols from 0 to {}".format(levels-1))
+
+    finished = False
+    result = True
+    # If no strength inputted, try increasing values for t until one
+    # does not work. If strength is inputted end after one check
+    while finished is False:
+        # Iterate over every possible selections of t columns, and
+        # count the t-tuples appearing in each selection
+        for comb in combinations(range(number_columns), wstrength):
+            tuple_dictionary = {item: 0 for item in product(symbol_list, repeat=wstrength)}
```

Comment:
Why do you need to count here? You just need to know that each combination appears, don't you? I would just create the set of row-combinations and check its size
```
existing_combinations = set(tuple([row[ti] for ti in comb]) for row in array)
if len(existing_combinations) != number_columns ** wstrength:
    ...
```